File: Javascript/Strings/D17_NonRepeatingCharacter.js
function nonRepeatingChar(s) {
      let freq = new Array(26).fill(0);

      for (let c of s) {
            freq[c.charCodeAt(0) - 97]++;
      }

      for (let c of s){
            if (freq[c.charCodeAt(0) - 97] === 1) {
                  return c;
            }
      }
      
      return '$';
}

console.log(nonRepeatingChar("geeksforgeeks"))
console.log(nonRepeatingChar("racecar"))
console.log(nonRepeatingChar("aabbccc"))

File: Java/String/D17_NonRepeatingCharacter.java
package String;

public class D17_NonRepeatingCharacter {
    public static void main(String[] args) {
        String s1 = "geeksforgeeks", s2 = "racecar", s3 = "aabbccc";
        System.out.println(nonRepeatingCharacter(s1));
        System.out.println(nonRepeatingCharacter(s2));
        System.out.println(nonRepeatingCharacter(s3));
    }
    public static char nonRepeatingCharacter(String s) {
        int[] frequency = new int[26];


        for (int i = 0 ; i < s.length(); i++) {
            frequency[s.charAt(i) - 'a']++;
        }

        for (int i = 0 ; i < s.length(); i++) {
            if (frequency[s.charAt(i) - 'a'] == 1) {
                return s.charAt(i);
            }
        }

        return '$';
    }
}
